### PR TITLE
Simplify matrix views

### PR DESCRIPTION
**Pull request by nimmolo**
This greatly simplifies the logic and html on `matrix_box` pages. 

I'd also like to get rid of the `matrix_table` layout (now called `matrix_grid`) because it is now superfluous, but i seem to need the logic of it for some reason on `reuse_image` and `remove_image` templates. 

If i try to bypass the call to `render(layout: "shared/matrix_grid", locals: {objects: @object.images}) do |image|` and directly render the block of html by iterating over each image, like this: 
```ruby
  @object.images.each do
    ...an html block with the image and a checkbox...
  end
```
... the loop outputs the images correctly, but then adds a "proxy" of the first **image record** at the end of the block, which is not intended, and rightly flagged as `unsafe_html`.

Suggestions welcome, it would be nice to solve this now, but not really necessary. This is already a big improvement on the current HTML for matrix box layouts, eliminating many useless divs.

### CODE REVIEW

File: app/views/image/_reuse_image.html.erb [hunk 1]
```diff
@@ -12,18 +12,25 @@
...
+      <%= @objects.each_with_index do |image, idx| %>
+        <li class="matrix-box col-xs-12 col-sm-6 col-md-4">
+          <div class="panel panel-default">
+            <div class="panel-body">
+              <%= thumbnail(image, link: url_args.merge(img_id: image.id), original: true) %>
+            </div>
+          </div>
+        </li>
+      <% end %>
+    </ul>
```
Comment (nimmolo) [inline app/views/image/_reuse_image.html.erb:34]: Why does this generate HTML with an `unsafe_html` error:
```ruby
      <%= @objects.each_with_index do |image, idx| %>
        <li class="matrix-box col-xs-12 col-sm-6 col-md-4">
          <div class="panel panel-default">
            <div class="panel-body">
              <%= thumbnail(image, link: url_args.merge(img_id: image.id), original: true) %>
            </div>
          </div>
        </li>
      <% end %>
```
But this does not?
```ruby
      <%= render(layout: "shared/matrix_table",
                 locals: {objects: @objects}) do |image| %>
        <div class="mt-3 mb-3">
          <%= thumbnail(image, link: url_args.merge(img_id: image.id), original: true) %>
        </div>
      <% end %>
```

**Comment by nimmolo** [comment]
@JoeCohen This is ready to go, too.

**Comment by JoeCohen** [comment]
Deployed.

File: app/views/image/_reuse_image.html.erb (hunk 1 above)
Comment (pellaea) [reply app/views/image/_reuse_image.html.erb:37]: That's an easy one!  I can answer that!  You're doing <%= each %> when you should only be doing <% each %>. :)

**Comment by pellaea** [comment]
Oops, looks like I'm late to the party.  I saw the review request and started, not realizing it had already been deployed.  Sorry!

**Comment by JoeCohen** [comment]
The comment was great. Now Nimmo can fix it.

File: app/views/image/_reuse_image.html.erb (hunk 1 above)
Comment (nimmolo) [reply app/views/image/_reuse_image.html.erb:37]: Agh! ❤️ You're the best! 
Had me completely flummoxed, researching the inner secrets of `yield`...

Will do a small PR to get this fixed

**Comment by JoeCohen** [comment]
Are you intending to leave the TODO
in  app/assets/javascripts/box_resizer.j?

**Comment by nimmolo** [comment]
Yes... Hoping to move to bootstrap 4 within the month. The resizer script is unnecessary with display: flex.

**Pull request merged**